apps/api/test/functional/templates.spec.ts

```diff
@@ -0,0 +1,110 @@
+import * as fs from "node:fs";
+import * as path from "node:path";
+
+import { app } from "@src/app";
+import type { GetTemplateByIdResponse, GetTemplatesFullResponse, GetTemplatesListResponse } from "@src/template/http-schemas/template.schema";
+import { env } from "@src/utils/env";
+
+const fakeHeaders = {
+  headers: {
+    "x-ratelimit-remaining": "100"
+  }
+};
+
+env.GITHUB_PAT = "ghp_1234567890";
+const sha = "cached-sha";
+
+jest.mock("@octokit/rest", () => {
+  return {
+    Octokit: jest.fn().mockImplementation(() => ({
+      rest: {
+        repos: {
+          getBranch: jest.fn().mockImplementation(() => {
+            return Promise.resolve({
+              status: 200,
+              data: {
+                commit: {
+                  sha
+                }
+              },
+              ...fakeHeaders
+            });
+          })
+        }
+      }
+    }))
+  };
+});
+
+describe("Templates API", () => {
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+
+  const expectCategory = (result: GetTemplatesFullResponse, expectedCategory: string, expectedTemplateIds: string[]) => {
+    const category = result.find((c: { title: string }) => c.title === expectedCategory);
+    expect(category.templates).toHaveLength(expectedTemplateIds.length);
+    const templateIds = category.templates.map(({ id }) => id);
+    templateIds.forEach(templateId => {
+      expect(expectedTemplateIds).toContain(templateId);
+    });
+  };
```

Comment:
suggestion(non-blocking): introducing reusable assertion function is bad idea because when test fails, test runner will display you part of the stack trace from reusable function which won't be informative.

Better to rewrite this into shorter form or create a custom expect matcher. For example:

```ts
expect(findCategory(result, expectedCategory)?.templates).toHaveLength(5);
expect(findCategory(result, expectedCategory)?.templates.map(t => t.id)).toEqual(expect.arrayContaining(expectedTemplateIds));
```